core/src/cf/mutations.rs

```diff
@@ -21,10 +21,11 @@ pub enum TableMutation {
 	Del(Thing),
 	Def(DefineTableStatement),
 	#[revision(start = 2)]
-	/// Includes the ID, current value, and changes that were applied to achieve this value
-	/// Example, ("mytb:tobie", {{"note": "surreal"}}, [{"op": "add", "path": "/note", "value": "surreal"}])
+	/// Includes the ID, current value (after change), changes that were applied to achieve this
+	/// value, and if this is a new record (i.e. create = true vs update = false)
+	/// Example, ("mytb:tobie", {{"note": "surreal"}}, [{"op": "add", "path": "/note", "value": "surreal"}], false)
 	/// Means that we have already applied the add "/note" operation to achieve the recorded result
-	SetWithDiff(Thing, Value, Vec<Operation>),
+	SetWithDiff(Thing, Value, Vec<Operation>, bool),
```

Comment:
This is quite minor but we should avoid bare booleans with non-obvious meaning as much a possible. Not blocking for this but I might suggest either creating another enum, or making it an inline struct with names.

Reply:
Yeah its a good point and agree with it thanks - will struct it

Reply:
Hey! What would be the reason for differentiating create and update?

It has been "Set" and not "Create" and "Update" because we don't differentiate between the two in terms of mutations made to the table/database, as unnecessary for CDC purposes.

At the higher level, LQ is a CDC framework for a view or a temporary table derived from the original table and its CF. So, I wonder if it's trivial to differentiate between create vs update for LQ.

At the lower level, a LQ user would need to receive the new state of the virtual doc/row in the view defined by the LQ, on each update made to the view.

Once the user receives the new state of virtual doc/row, they need to update the locally cached version of the view(either directly or indirectly, depending on how the user implemented their own app) with the new state, regardless of whether it was from a create or an update, thus making this information about create vs update useless for LQ users? 🤔

Reply:
I have changed this to use Set and SetWithDiff

RE usage of LQ, thats how I imagine it yes. Users do need to differentiate client side as they are building a snapshot from events.